openfisca_us/variables/demographic/household.py

```diff
@@ -32,3 +33,104 @@ class is_household_head(Variable):
     def formula(person, period, parameters):
         # Use order of input (first)
         return person.household.members_position == 0
+
+
+class State(Enum):
+    # Source: https://gist.github.com/rogerallen/1583593#gistcomment-3699885
+    # 50 states and DC.
+    AL = "Alabama"
+    AK = "Alaska"
+    AZ = "Arizona"
+    AR = "Arkansas"
+    CA = "California"
+    CO = "Colorado"
+    CT = "Connecticut"
+    DE = "Delaware"
+    FL = "Florida"
+    GA = "Georgia"
+    HI = "Hawaii"
+    ID = "Idaho"
+    IL = "Illinois"
+    IN = "Indiana"
+    IA = "Iowa"
+    KS = "Kansas"
+    KY = "Kentucky"
+    LA = "Louisiana"
+    ME = "Maine"
+    MD = "Maryland"
+    MA = "Massachusetts"
+    MI = "Michigan"
+    MN = "Minnesota"
+    MS = "Mississippi"
+    MO = "Missouri"
+    MT = "Montana"
+    NE = "Nebraska"
+    NV = "Nevada"
+    NH = "New Hampshire"
+    NJ = "New Jersey"
+    NM = "New Mexico"
+    NY = "New York"
+    NC = "North Carolina"
+    ND = "North Dakota"
+    OH = "Ohio"
+    OK = "Oklahoma"
+    OR = "Oregon"
+    PA = "Pennsylvania"
+    RI = "Rhode Island"
+    SC = "South Carolina"
+    SD = "South Dakota"
+    TN = "Tennessee"
+    TX = "Texas"
+    UT = "Utah"
+    VT = "Vermont"
+    VA = "Virginia"
+    WA = "Washington"
+    WV = "West Virginia"
+    WI = "Wisconsin"
+    WY = "Wyoming"
+    DC = "District of Columbia"
+    # Territories.
+    GU = "Guam"
+    MP = "Northern Mariana Islands"
+    PW = "Palau"
+    PR = "Puerto Rico"
+    VI = "Virgin Islands"
+    AA = "Armed Forces Americas (Except Canada)"
+    AE = "Armed Forces Africa/Canada/Europe/Middle East"
+    AP = "Armed Forces Pacific"
+
+
+class state_code(Variable):
+    value_type = Enum
+    possible_values = State
+    default_value = State.CA
+    entity = Household
+    label = u"State"
+    definition_period = ETERNITY
+
+
+class StateGroup(Enum):
+    CONTIGUOUS_US = "Contiguous US"
+    AK = "Alaska"
+    HI = "Hawaii"
+    GU = "Guam"
+    PR = "Puerto Rico"
+    # Omit other territories for now.
```

Comment:
The CPS only has the 50 states and DC. If any new ones show up, the current logic will slot them into CONTIGUOUS_US. To avoid that, we'd have to make a list of all the states, which I guess we could do but ideally I think we'd use something like the us package to automate it.